File: workspace/silnia.py
# from silnia.cpp to silnia.py

def IterateFactorial(n):
    suma = 1
    i = 1
    while i <= n:
        suma *= i
        i += 1
    return suma


def RecursiveFactorial(n):
    if n == 0 or n == 1:
        return 1
    else:
        return n * RecursiveFactorial(n-1)


def main():
    n = 6
    print("Iteracyjnie ", IterateFactorial(n), '\n', sep='', end='')
    print("Rekurencyjnie ", RecursiveFactorial(n), sep='', end='')
    return 0


if __name__ == '__main__':
    main()

File: workspace/silnia.cpp
#include<iostream>
using namespace std;
int IterateFactorial(int n)
{
	int suma =1;
	for(int i=1;i<=n;i++)
	{
		suma *=i;
	}
	return suma;
}
int RecursiveFactorial(int n)
{
	if (n==0 || n==1)
	{
		return 1;
	}
	else
	{
		return n*RecursiveFactorial(n - 1);
	}
}
int main()
{
	int n=6;
	cout<<"Iteracyjnie "<<IterateFactorial(n)<<endl;
	cout<<"Rekurencyjnie "<<RecursiveFactorial(n);
	return 0;
}